ddp-workspace/projects/ddp-dsm-ui/src/app/scan/scan.component.ts

```diff
@@ -139,6 +139,7 @@ export class ScanComponent implements OnInit {
   private addNewScanPair(): void {
     const newRow = new ScanPairComponent();
```

Comment:
Why are we instantiating a new Angular component here?

Reply:
That was there since before this PR. I believe line 140 is what adds another input field to the UI. So after the first label or short id is filled out the second one will appear because of this function. Does this answer your question @GiCharkviani ?

Reply:
I see. It just doesn't make sense to instantiate a new Angular component (which is a way bigger thing than just a class) only for filling the empty array.
+ As far as I investigated, we don't even use the values stored in the 'scanPairs', which means that even if you pushed strings or numbers it would be just fine :D 

I would just suggest slowly, whenever you can, just to improve the code a little bit.

Reply:
It's just my opinion, do as you wish.

Reply:
@ngigliotosBroad